cardhost/app/lib/builder.ts

```diff
@@ -49,13 +82,27 @@ export default class Builder implements BuilderInterface {
   async getBuiltCard(
     url: string,
     format: 'isolated' | 'embedded'
-  ): Promise<{ model: any; componentImplementation: unknown }> {
+  ): Promise<{ model: any; moduleName: string }> {
     let compiledCard = await this.getCompiledCard(url);
-    let templateSource = compiledCard.templateSources[format];
-    let componentImplementation = setComponentTemplate(
-      compileTemplate(templateSource),
-      templateOnlyComponent()
-    );
-    return { model: compiledCard.data, componentImplementation };
+    // let templateSource = await window.require(
+    //   compiledCard.templateModules[format].moduleName
+    // );
+    // let componentImplementation = setComponentTemplate(
+    //   compileTemplate(templateSource),
+    //   templateOnlyComponent()
+    // );
+    // this.defineModule(compiledCard.modelModule, templateSource);
+
+    /*
+      TODO Build stage:
+        1. Require template module from it's path, eg: "http://mirage/cards/hello/isolated"
+        2. Babel transpile it
+        3. Do the createTemplateFactory song and dance?
+        4. Redefine the module? Make a new definintion?
+    */
```

Comment:
@ef4 This is the part that I'm not stuck on. 

1. I'm not quite sure how we want to handle the module resolution and definition for these built modules
2. I'm not sure of the full scope of babel plugins we'll need and how to configure them. Do you think we'll just copy the logic from here? https://github.com/embroider-build/embroider/blob/master/packages/core/src/app.ts#L337
3. Apart from the babel config, will we also need to wrap the components in the templateFactory stuff?